src/utils/hooks.js

```diff
@@ -19,3 +19,61 @@ export function useStateWithStorage(storageKey, initialValue) {
 
 	return [value, setValue];
 }
+
+export function useVoiceToText() {
+	const [text, setText] = useState('');
+	const [isListening, setIsListening] = useState(false);
+	const recognitionRef = useRef(null);
+
+	useEffect(() => {
+		if (
+			!('SpeechRecognition' in window || 'webkitSpeechRecognition' in window)
+		) {
+			console.error('SpeechRecognition is not supported by this browser.');
```

Comment:
**issue (blocking)** When I try the feature in safari, I'm noticing that when I give permissions for my mic with `allow` It will freeze not at `Listening...` and the only way to get out of it is to refresh or close out of the tab. And there are no errors in the console. 

And when I press `deny` for mic permissions it goes immediately back to `Start Listening` and I see the errors in the console every click, so all good there. So, I'm guessing it's not accounting for it if the user gives mic permissions? Maybe you could use the [`MediaDevices`](https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices) `getUserMedia()` method.